### PR TITLE
[device_mesh][BE] reduce_scatter fallback to funcol and remove from DM

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #105642
* #105614
* #105605
* #105604

For the reason similar to https://github.com/pytorch/pytorch/pull/105605

### CODE REVIEW

File: torch/distributed/_tensor/placement_types.py [hunk 1]
```diff
@@ -199,8 +199,8 @@ def _reduce_shard_tensor(
             )
             tensor = torch.cat(scattered_list, dim=self.dim)
 
-        output = mesh.reduce_scatter(
-            tensor, op=reduce_op, mesh_dim=mesh_dim, scatter_dim=self.dim
+        output = funcol.reduce_scatter_tensor(
+            tensor, reduce_op.name, scatter_dim=self.dim, group=(mesh, mesh_dim)
```
Comment (kumpera) [inline torch/distributed/_tensor/placement_types.py:203]: Should we support reduce_op directly as well in the python wrappers of funcol?

**Review by kumpera — APPROVED** [review]
LGTM

**Review by wz337 — APPROVED** [review]
LGTM!

**Review by fduwjj — APPROVED** [review]

File: torch/distributed/_tensor/placement_types.py (hunk 1 above)
Comment (wanchaol) [reply torch/distributed/_tensor/placement_types.py:203]: yeah that's a good point.. I can give it a try in a follow up PR

**Pull request closed without merging**